lib/LaTeXML/Package/TeX.pool.ltxml

```diff
@@ -243,7 +243,9 @@ DefParameterType('DefExpanded', sub {
     local $LaTeXML::KEEP_THE = 1;
     my $token    = $gullet->readXToken;
     my $expanded = ($token->getCatcode == CC_BEGIN ? $gullet->readBalanced(1) : $token);
-    return $expanded->without_dont_expand; },
+    $expanded = Tokens(PrepArgTokens(@$expanded))->without_dont_expand if $expanded;
```

Comment:
Just what I was thinking of!  But in my experiments, I followed by removing all NOEXPAND1's to limit their scope (& similar in a few other places).  Is that a good strategy? It seems your strategy is to try to force a fail if any leak past?

Reply:
I am also trying to remove them everywhere I can, but I try to do it in a TeXy way. As in:
 - they will be removed automatically by `readToken` and `readXToken` whenever `KEEP_THE` is 0
 - when `KEEP_THE` is 1, they will get carried out of Gullet to the two parameters `DefPlain` and `DefExpanded`.
 - those two parameters need to always invoke `->without_dont_expand` on their final tokens, ensuring the flags never make it to the Expandable definition.

There should be nothing else to remove from a TeX perspective, but any future code that decides to use `KEEP_THE=1` (if any) will have to be alert to not leak anything through. And if they do, getting errors quickly will be a nice bit of help.

Reply:
Actually, I am now realizing this is extra safe (and hence extra slow), since `PrepArgTokens` is itself already calling `->without_dont_expand` on the rescanned tokens it returns. So calling that _early_ is all one needs to do when `KEEP_THE` is set to 1, while nothing extra should be needed when `KEEP_THE` is 0.